core/node/groups.go

```diff
@@ -133,17 +134,30 @@ func LibP2P(bcfg *BuildCfg, cfg *config.Config) fx.Option {
 			logger.Error("'Swarm.EnableAutoRelay' has been overridden by 'Swarm.AutoRelay.Enabled'")
 		}
 	}
-	//nolint
+	// nolint
 	if cfg.Swarm.EnableRelayHop {
 		logger.Fatal("The `Swarm.EnableRelayHop` config field is ignored.\n" +
 			"Use `Swarm.RelayService` to configure the circuit v2 relay.\n" +
 			"If you want to continue running a circuit v1 relay, please use the standalone relay daemon: https://github.com/libp2p/go-libp2p-relay-daemon (with RelayV1.Enabled: true)")
 	}
 
+	// Swarm.ResourceMgr
+	enableResourceMgr := cfg.Swarm.ResourceMgr.Enabled.WithDefault(false)
+	/// ENV overrides Config (if present)
+	switch os.Getenv("IPFS_RCMGR") {
+	case "0", "false":
+		enableResourceMgr = false
+	case "1", "true":
+		enableResourceMgr = true
+	}
+
 	// Gather all the options
 	opts := fx.Options(
 		BaseLibP2P,
 
+		// Services (resource management)
+		maybeProvide(libp2p.ResourceManager(cfg.Swarm.ResourceMgr), enableResourceMgr),
```

Comment:
@marten-seemann @vyzo  is it possible to run go-libp2p >=0.18 without _any_ resource manager?

- This line will skip RM registration when `enableResourceMgr` is `false` (default), but I am unsure if there are any implicit defaults in go-libp2p.
  - If so, I can change this to always initialize but use `network.NullResourceManager` instead (like in https://github.com/filecoin-project/lotus/pull/8318)

Reply:
no, you need to pass the nullman to disable.